# Assignment 3: 

## Professor demonstration, with my contribution starting at the 'To Do' section

### Set working environment

In [4]:
import os
import sys
import numpy as np
import pylab as pl


import json
import os

%pylab inline


Populating the interactive namespace from numpy and matplotlib


### Null Hypothesis: the % of former prisoners employed after release is the same or lower for candidates who participated in the program as for the control group, significance level p=0.05

$H_0: P_0 - P_1 \geq$ 0

$H_a: P_0 - P_1 $< 0

$\alpha$ = 0.05

### this is a TEST OF PROPORTIONS. we use the Binomial distribution since it is a yes/no (bernulli) test for each subject: the former inmate was or was not ever employed in a CEO transitional job (second row in the table above):

$P_0=0.035

P_1=0.701$

In [5]:
#set significance level before performing calculations:
alpha=0.05

P_0 = 3.5/100 
P_1 = 70.1/100

if P_0-P_1 >= 0:
    # we are done
    print ("the Null holds")
else:
    print ("we must assess the statistical significance")

n_0 = 409
n_1 = 564

#find counts by multiplying sample size:
Nt_0 = P_0 * n_0
Nt_1 = P_1 * n_1


we must assess the statistical significance


### Perform Z Test:

In [6]:
#define sample standard deviation:

#first find the sample proportion
sp = (P_0 * n_0 + P_1 * n_1) / (n_1 + n_0)
print (sp)

#find standard deviation
sp_stdev = lambda p, n: np.sqrt( p * ( 1 - p ) / n[0] +  p * ( 1 - p ) / n[1]  )


sp_stdev_2y = sp_stdev(( Nt_0 + Nt_1) / (n_0 + n_1), [n_0, n_1])
print (P_0, n_0, n_1, sp_stdev_2y)

0.421047276465
(0.035, 409, 564, 0.03206580860573819)


In [7]:
#find z score:
zscore = lambda p0, p1, s : (p0 - p1) / s
z_2y = zscore(P_1, P_0, sp_stdev_2y)
print (z_2y)

20.7697865408


### Find P-Value

In [8]:
#find p-value using table:

# p-value for employment after 2 years: 
# since the largest number we read off the table for alpha=0.05 is smaller then the value for our statistic 
# our p-value will be smaller then calculated using .9998

p_2y = 1 - 0.9998


def report_result(p,a):
    print ('is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p,a))
    if p < a:
        print ("YES!")
    else: 
        print ("NO!")
    
    print ('the Null hypothesis is {}'.format( 'rejected' if p < a  else 'not rejected') )

    
report_result(p_2y,alpha)

is the p value 0.00 smaller than the critical value 0.05? 
YES!
the Null hypothesis is rejected


### Conclusions:

The p-value, 0.00, was smaller than the critical value 0.05 (significance level). Therefore, the Null Hypothesis is rejected, which means the % of former prisoners employed after release is greater for candidates who participated in the program as for the control group. This also means that it is not by chance alone and it is statistically significant.

# To Do:

### Former inmate was or was not "Convicted of a felony" (row 10) in the Recidivism (Years 1-3)

### Null hypothesis:
The % of former prisoners convicted of a felony is the same or higher for candidates who participated in the program as for the control group

significance level p=0.05

H0:P0f−P1f≥ 0 

Ha:P0f−P1f< 0 

α = 0.05


P0f=0.117,P1f=0.100

In [16]:
#set significance level before performing calculations:
#define variables used in this analysis:
alpha=0.05

P_0f = 11.7/100 
P_1f = 10.0/100

if P_0f-P_1f >= 0:
    # we are done
    print ("the Null holds")
else:
    print ("we must assess the statistical significance")

n_0f = 409
n_1f = 568

#find counts by multiplying sample size:
Nt_0f = P_0f * n_0f
Nt_1f = P_1f * n_1f


the Null holds


### Z Test:

In [17]:
#define sample standard deviation:

#first find the sample proportion
spf = (P_0f * n_0f + P_1f * n_1f) / (n_1f + n_0f)
print (spf)

#find standard deviation
sp_stdevf = lambda pf, nf: np.sqrt( pf * ( 1 - pf ) / nf[0] +  pf * ( 1 - pf ) / nf[1]  )


sp_stdev_2yf = sp_stdevf(( Nt_0f + Nt_1f) / (n_0f + n_1f), [n_0f, n_1f])
print (P_0f, n_0f, n_1f, sp_stdev_2yf)

0.107116683726
(0.11699999999999999, 409, 568, 0.020055679161170879)


In [18]:
#find z score:
zscoref = lambda p0f, p1f, sf : (p0f - p1f) / sf
z_2yf = zscoref(P_1f, P_0f, sp_stdev_2yf)
print (z_2yf)

-0.84764020522


In [14]:
#find p-value using table:

# p-value for employment after 2 years: 
# since the largest number we read off the table for alpha=0.05 is smaller then the value for our statistic 
# our p-value will be smaller then calculated using .9998

p_2yf = 1 - 0.8023


def report_result(pf,af):
    print ('is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(pf,af))
    if pf < af:
        print ("YES!")
    else: 
        print ("NO!")
    
    print ('the Null hypothesis is {}'.format( 'rejected' if pf < af  else 'not rejected') )

    
report_result(p_2yf,alpha)

is the p value 0.20 smaller than the critical value 0.05? 
NO!
the Null hypothesis is not rejected


### Conclusions:

The Null Hypothesis was not rejected in the Z-test above, therefore the difference is not statistically significant.

## Chi Squared Analysis:

In [34]:
chisqstat= lambda N, values, expect_num : ((values[0][0] * values[1][1]
                                            -values[0][1] * values[1][0])**2) * N / expect_num
Ntot = 977 # a + b + c + d = tot

tot1 = ((n_0f*P_0f)+(n_1f*P_1f))
tot2 = ((n_0f*(1-P_0f))+(n_1f*(1-P_1f)))



expected_num = n_0f * n_1f * tot1 * tot2
sample_values = [[P_0f * n_1f, (1-P_0f) * n_1f], [P_1f * n_0f, (1-P_1f) * n_0f]]
 

print (chisqstat(Ntot,  sample_values, expected_num))

0.718493917505


Degrees of Freedom: 1

### Comparison to Chi-Squared Distribution and Conclusions:

0.7185 is less than 3.84 so therefore we do not reject the Null Hypothesis. Again, the difference between variables in not statistically signifcant in this case.

In [35]:
print(n_0f)
print(n_1f)

409
568


In [36]:
print(P_0f)
print(P_1f)

0.117
0.1


In [43]:
print(tot1)
print(tot2)

104.653
872.347


### Contigency Table:

|Convicted of a felony |Employed |Not Employed |
|:----------:||:----------:||:----------:|
|test sample|409 * 0.117|409 * (1-0.117)|
|control sample|568 * 0.1|568 * (1-0.1)|
|total|104.653|872.347|